File: src/ast/tree/statement/break.ts
import type { Pass } from "../../../pass/pass";
import type { Block } from "../../node/block";
import { ExecutionResult, Statement } from "../../node/statement";

export class Break extends Statement {
    constructor(parent: Block | null) {
        super(parent);
    }

    public execute(): ExecutionResult {
        return ExecutionResult.BreakLoop;
    }

    public visit(pass: Pass): void {
        pass.visitBreak(this);
    }
}

File: src/main/kotlin/me/honkling/miniscript/parser/ast/expression/Break.kt
package me.honkling.miniscript.parser.ast.expression

import me.honkling.miniscript.parser.ast.Block
import me.honkling.miniscript.parser.ast.Node
import me.honkling.miniscript.parser.ast.statement.ExecutionResult
import me.honkling.miniscript.parser.ast.statement.Statement
import me.honkling.miniscript.pass.Pass

class Break(parent: Node<*>?) : Expression<Nothing?>(parent) {
    override fun get(): Pair<Nothing?, ExecutionResult> {
        return null to ExecutionResult.BreakLoop
    }

    override fun accept(pass: Pass) {
        pass.visitBreak(this)
    }
}